src/main/scala/com/webtrends/harness/component/kafka/actor/OffsetManager.scala

```diff
@@ -45,46 +100,83 @@ with ActorLoggingAdapter with ZookeeperAdapter {
    */
   def getOffsetState(req: GetOffsetData) = {
     val originalSender = sender()
-    val parent = context.parent
-    //Path to write to zookeeper
-    val path = s"$offsetPath/${req.path}"
-    getData(path).onComplete {
-      case Success(data) =>
-        Try {
-          originalSender ! OffsetDataResponse(Left(OffsetData(data)))
-          parent ! healthy(path)
-        } recover {
-          case ex =>
-            log.error("Error retrieving state from zk", ex)
-            originalSender ! OffsetDataResponse(Right(ex))
-            parent ! unhealthy(path)
-        }
-      case Failure(err) =>
-        err match {
-          case ex: NoNodeException =>
-            originalSender ! OffsetDataResponse(Left(OffsetData(Array.empty[Byte])))
-            parent ! healthy(path)
-          case ex =>
-            log.error("Unable to get state from Zk", ex)
-            originalSender ! OffsetDataResponse(Right(ex))
-            parent ! unhealthy(path)
+
+    brokers.get(req.cluster) match {
+      case Some(brokerInfo) =>
+        val channel = brokerInfo._1
+        val broker = brokerInfo._3
+        try {
+          val partitions = List(new TopicAndPartition(req.topic, req.partition))
+          val fetchRequest = new OffsetFetchRequest(pod, partitions, 1, brokerInfo._2, clientId)
+
+          channel.send(fetchRequest.underlying)
```

Comment:
Question which offset manager are you talking to? It looks like the source brokers?